code42/py42
# add support for 2FA

### Description of Change ###

This change adds support for Code42's two-factor authentication implementation using time-based one-time passwords

### Issues Resolved ###

- closes #195

### Testing Procedure ###
1. Create a Code42 account in an organization that has two-factor authentication turned on
2. Attempt to create a SDK object using this account
3. Verify that Py42MFARequiredError is raised
4. Attempt to create a SDK object and include the TOTP token (`sdk.from_local_account(hostaddress,username,password,totp="123456"`)
5. Verify that the SDK object is created successfully

### PR Checklist ###
Did you remember to do the below?


I have not added tests yet as I am not sure how best to do so. I would appreciate recommendations/input.
